lib/IRGen/GenMeta.cpp

```diff
@@ -3858,11 +3868,14 @@ namespace {
     // Visitor methods.
 
     void addValueWitnessTable() {
+      if (IGM.getOptions().LazyInitializeClassMetadata)
+        return B.addNullPointer(IGM.WitnessTablePtrTy);
```

Comment:
I don't think the runtime will actually initialize the value witness table in the case where you emit null here?

Reply:
Oh, hmm, what needs to be changed to have that be initialised?

Reply:
Something in the runtime

Reply:
Any hints as to where to start making that change?

Reply:
Hmm, so, running something like this:

```swift
import CoreFoundation
public class NSString {
  var cfObj: CFString?
}
```
does provide the completion function which is initialising the VWT:
```llvm
define internal swiftcc %swift.metadata_response @"$sSo11CFStringRefaMr"(%swift.type*, i8*, i8**) #0 {
entry:
  %3 = bitcast %swift.type* %0 to %swift.type**
  %4 = getelementptr inbounds %swift.type*, %swift.type** %3, i32 2
  store %swift.type* getelementptr inbounds (%swift.full_type, %swift.full_type* @"$sBoN", i32 0, i32 1), %swift.type** %4, align 8
  ret %swift.metadata_response zeroinitializer
}
```

Reply:
That doesn't look like it's initializing the VWT.  Looks like it's initializing the superclass pointer to the  metadata for `Builtin.NativeObject`, which doesn't make any sense.

Reply:
Thanks @rjmccall!  I've modified the runtime now to initialize the VWT as part of the cache setup as suggested by you below.